modules/phase_field/src/materials/MathFreeEnergy.C

```diff
@@ -0,0 +1,50 @@
+#include "MathFreeEnergy.h"
+
+template<>
+InputParameters validParams<MathFreeEnergy>()
+{
+  InputParameters params = validParams<DerivativeBaseMaterial>();
+  params.addClassDescription("Material that implements the math free energy and its derivatives: \nF = 1/4(1 + c)^2*(1 - c)^2");
+  params.addRequiredCoupledVar("c","Concentration variable");
+  return params;
+}
+
+MathFreeEnergy::MathFreeEnergy(const std::string & name,
+                       InputParameters parameters) :
+    DerivativeBaseMaterial(name, parameters),
+    _c(coupledValue("c")),
+    _c_var(coupled("c"))
+{}
+
+Real
+MathFreeEnergy::computeF()
+{
+  return 1.0/4.0*(1.0 + _c[_qp])*(1.0 + _c[_qp])*(1.0 - _c[_qp])*(1.0 - _c[_qp]);
+}
+
+Real
+MathFreeEnergy::computeDF(unsigned int j_var)
+{
+  if (j_var == _c_var)
```

Comment:
P.S. your code accidentally works if you have only one variable in your problem, or pass the variables in the correct order.